src/filters/webgl_backend.class.ts

```diff
@@ -288,14 +312,45 @@ class WebglFilterBackend {
    *
    * @param {String} cacheKey The cache key provided when the source image was filtered.
    */
-  evictCachesForKey(cacheKey) {
+  evictCachesForKey(cacheKey: string) {
     if (this.textureCache[cacheKey]) {
       this.gl.deleteTexture(this.textureCache[cacheKey]);
       delete this.textureCache[cacheKey];
     }
   }
 
-  copyGLTo2D: copyGLTo2DDrawImage,
+  /**
+   * Copy an input WebGL canvas on to an output 2D canvas.
+   *
+   * The WebGL canvas is assumed to be upside down, with the top-left pixel of the
+   * desired output image appearing in the bottom-left corner of the WebGL canvas.
+   *
+   * @param {WebGLRenderingContext} sourceContext The WebGL context to copy from.
+   * @param {Object} pipelineState The 2D target canvas to copy on to.
+   */
+  copyGLTo2D(gl: WebGLRenderingContext, pipelineState: TWebGLPipelineState) {
```

Comment:
This method can be static
I saw the code calls it with a call directive: `copyGLTo2D.call(context, ...params)` but there is no need sine it is context less

Reply:
Ahh I saw the assignment of `copyGLTo2DPutImageData` that uses context

Why are those methods standalone and `copyGLTo2D` is a class method?

Reply:
Perhaps create a dedicated lass for these methods, passing the context to the constructor?
Instead of using call and casting types, perf-wise is identical I believe

Reply:
yeah this method swap here isn't a good idea, but for now i m not going to touch it